### PR TITLE
working on sml logger sample

### PR DESCRIPTION
**Pull request by magni-mar**

### CODE REVIEW

File: libs/dbus_util/tests/examples/example_sml_interface.cpp [hunk 1]
```diff
@@ -1,25 +1,61 @@
...
+  std::shared_ptr<sdbusplus::asio::dbus_interface> const interface {
+    std::make_shared<sdbusplus::asio::dbus_interface>(ipc_client.connection(),
+                                                      std::string{ tfc::dbus::sml::tags::path },
+                                                      tfc::dbus::make_dbus_name("example_state_machine"))
+  };
+
   tfc::dbus::sml::interface sml_interface {
-    interface, "Log key"
-  };  // optional log key
-  // NOTE! interface struct requires to be passed by l-value like below, so the using code needs to store it like above
```
Comment (jbbjarnason) [inline libs/dbus_util/tests/examples/example_sml_interface.cpp:22]: I think this text should not be removed.

**Review by jbbjarnason — CHANGES_REQUESTED** [review]
Please update the example in the header file along with these changes

**Comment by magni-mar** [comment]
> Please update the example in the header file along with these changes

Sample in header file updated, please see changes.

File: libs/dbus_util/tests/examples/example_sml_interface.cpp [hunk 2]
```diff
@@ -1,25 +1,61 @@
...
-int main() {
+
+auto main(int argc, char** argv) -> int {
+  tfc::base::init(argc, argv);
   boost::asio::io_context ctx{};
-  auto bus = std::make_shared<sdbusplus::asio::connection>(ctx);
-  auto interface =
-      std::make_shared<sdbusplus::asio::dbus_interface>(bus, std::string{ tfc::dbus::sml::tags::path }, "StateMachineName");
+
+  tfc::ipc_ruler::ipc_manager_client const ipc_client{ ctx };
```
Comment (jbbjarnason) [inline libs/dbus_util/tests/examples/example_sml_interface.cpp:37]: I would prefer to use bare bone sdbusplus asio connection, this should only be minimal example

File: libs/dbus_util/tests/examples/CMakeLists.txt [hunk 3]
```diff
@@ -1,6 +1,11 @@
-
 add_executable(example_sml_interface example_sml_interface.cpp)
 
-target_link_libraries(example_sml_interface PRIVATE tfc::logger tfc::base tfc::dbus_util)
+target_link_libraries(example_sml_interface
+  PRIVATE
+    tfc::logger
+    tfc::base
+    tfc::dbus_util
+    tfc::ipc
```
Comment (jbbjarnason) [inline libs/dbus_util/tests/examples/CMakeLists.txt:8]:
```suggestion
```

File: libs/dbus_util/tests/examples/example_sml_interface.cpp [hunk 4]
```diff
@@ -1,25 +1,61 @@
...
 #include <sdbusplus/asio/object_server.hpp>
+
+#include <tfc/ipc.hpp>
+#include <tfc/progbase.hpp>
+
 #include <tfc/dbus/sml_interface.hpp>
-struct state_machine {
+#include <tfc/dbus/string_maker.hpp>
+
+struct running {
```
Comment (jbbjarnason) [inline libs/dbus_util/tests/examples/example_sml_interface.cpp:15]: I would say it is conflicting to call state running and event running, event can be named run and event stop

File: libs/dbus_util/tests/examples/example_sml_interface.cpp (hunk 2 above)
Comment (magni-mar) [reply libs/dbus_util/tests/examples/example_sml_interface.cpp:37]: changing to sdbusplus asio connection but a comment which specifies that `ipc_client` can also be used.

File: libs/dbus_util/tests/examples/example_sml_interface.cpp (hunk 4 above)
Comment (magni-mar) [reply libs/dbus_util/tests/examples/example_sml_interface.cpp:15]: Changing event names to `run` and `stop`

File: libs/dbus_util/tests/examples/example_sml_interface.cpp [hunk 5]
```diff
@@ -1,25 +1,61 @@
...
+
+auto main(int argc, char** argv) -> int {
+  tfc::base::init(argc, argv);
   boost::asio::io_context ctx{};
-  auto bus = std::make_shared<sdbusplus::asio::connection>(ctx);
-  auto interface =
-      std::make_shared<sdbusplus::asio::dbus_interface>(bus, std::string{ tfc::dbus::sml::tags::path }, "StateMachineName");
+
+  /// Raw dbus connection, ipc_client also has a dbus connection which can be used through ipc_client.connection()
+  std::shared_ptr<sdbusplus::asio::connection> const dbus_connection{ std::make_shared<sdbusplus::asio::connection>(ctx) };
```
Comment (jbbjarnason) [inline libs/dbus_util/tests/examples/example_sml_interface.cpp:37]: I would prefer auto but that's really nit-picking

File: libs/dbus_util/tests/examples/example_sml_interface.cpp [hunk 6]
```diff
@@ -1,25 +1,61 @@
...
+  std::shared_ptr<sdbusplus::asio::dbus_interface> const interface {
+    std::make_shared<sdbusplus::asio::dbus_interface>(dbus_connection,
+                                                      std::string{ tfc::dbus::sml::tags::path },
+                                                      tfc::dbus::make_dbus_name("example_state_machine"))
+  };
+
   tfc::dbus::sml::interface sml_interface {
     interface, "Log key"
   };  // optional log key
   // NOTE! interface struct requires to be passed by l-value like below, so the using code needs to store it like above
```
Comment (jbbjarnason) [inline libs/dbus_util/tests/examples/example_sml_interface.cpp:48]: This applies to line 52, move to there. Afterwards squash and merge

**Review by jbbjarnason — APPROVED** [review]
At least move comment, then merge

**Pull request merged**